Python/peephole.c

```diff
@@ -152,6 +152,13 @@ fold_tuple_on_constants(_Py_CODEUNIT *codestr, Py_ssize_t codelen,
         PyTuple_SET_ITEM(newconst, i, constant);
     }
 
+    Py_ssize_t index = PyList_GET_SIZE(consts);
+    if ((size_t)index >= UINT_MAX - 1) {
```

Comment:
Maybe add `#if SIZEOF_SIZE_T > 4` as in #11011? This condition always false on 32-bit platform.

See also #6545 which makes this change unneeded.

Reply:
"(size_t)index >= UINT_MAX" can be always true if size_t and unsigned int have the same size, but I use "(size_t)index >= UINT_MAX - 1"  which is different. Are you are that an #if is needed?

Reply:
index <= PY_SSIZE_T_MAX / sizeof(PyObject*) which is equal to `2**29-1` on 32-bit platform.

Reply:
Ok Ok, I added "#if SIZEOF_SIZE_T > SIZEOF_INT".